### PR TITLE
Don't pass all Galaxy environment variables to jobs

### PR DESCRIPTION
**Pull request by nsoranzo**
Some env variables may affect reproducibility, e.g. `LANG` changes the behaviour of tools like `sort`. Also tools may see information they should not have access to.

Also:
- Check that jobs don't inherit env variables from Galaxy in integration tests.
- Merge `RunsEnvironmentJobs` class into `BaseJobEnvironmentIntegrationTestCase`
- Remove redundant checks in `DefaultJobEnvironmentIntegrationTestCase`

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Review by hexylena — APPROVED** [review]

File: test/integration/test_job_environments.py
Comment (nsoranzo) [inline test/integration/test_job_environments.py]: This refers to https://github.com/galaxyproject/pulsar/pull/295 and would require also an update to `pulsar-galaxy-lib` requirement.

**Comment by nsoranzo** [comment]
This is now ready to re-review/merge.

File: test/integration/test_job_environments.py
Comment (mvdbeek) [reply test/integration/test_job_environments.py]: Let's wait for the release ? @jmchilton could you add me (and @nsoranzo if he wishes) to the pulsar maintainers on pypi, then I'm happy to cut a new release.

Comment (nsoranzo) [reply test/integration/test_job_environments.py]: Fine by me to wait for the new `pulsar-galaxy-lib` release. I am still not very familiar with pulsar's code base, so no rush to add me to its PyPI maintainers.

Comment (jmchilton) [reply test/integration/test_job_environments.py]: @mvdbeek you were on the list I think, at least PyPI says you were. I did add @nsoranzo now though.

I'll do a Pulsar release right now - it is a little different than the other projects since it releases two projects at once.

Comment (jmchilton) [reply test/integration/test_job_environments.py]: Release pushed and tagged.

**Review by jmchilton — APPROVED** [review]
It seems like it should be off during development by default for things like testing locally configured software but... I appreciate the advantages this approach provides.

File: test/integration/test_job_environments.py
Comment (mvdbeek) [inline test/integration/test_job_environments.py]:
```suggestion
            # Check that the job hasn't inherited env variables from Galaxy
            assert job_env_properties.some_env != "Value from env"
```

Comment (nsoranzo) [reply test/integration/test_job_environments.py]: Thanks, fixed in a commit where I've also updated pulsar-galaxy-lib to the latest version, and rebased.

File: test/integration/test_job_environments.py
Comment (nsoranzo) [reply test/integration/test_job_environments.py]: Thanks @jmchilton !

Comment (mvdbeek) [reply test/integration/test_job_environments.py]: Yes, thank you John! I bugged Nate about getting added to PyPI.

**Pull request merged**
